test/resources/knativeserving.go

```diff
@@ -36,20 +38,53 @@ import (
 
 const (
 	interval = 10 * time.Second
-	timeout  = 2 * time.Minute
+	timeout  = 5 * time.Minute
 )
 
+// WaitKnativeServingReady polls the status of the KnativeServing called name
+// from client every `interval` until `inState` returns `true` indicating it
+// is done, returns an error or timeout.
+func WaitKnativeServingReady(t *testing.T, clients *test.KnativeServingAlphaClients, name string,
+	inState func(s *v1alpha1.KnativeServing, err error) (bool, error)) (*v1alpha1.KnativeServing, error) {
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForKnativeServingState/%s/%s", name, "KnativeServingIsReady"))
+	defer span.End()
+
+	var lastState *v1alpha1.KnativeServing
+	waitErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		lastState, err := clients.KnativeServings.Get(name, metav1.GetOptions{})
+		return inState(lastState, err)
+	})
+
+	if waitErr != nil {
+		return lastState, errors.Wrapf(waitErr, "knativeserving %s is not in desired state, got: %+v", name, lastState)
+	}
+	return lastState, nil
+}
+
+// CreateKnativeServing creates a KnativeServing with the name names.KnativeServing under the namespace names.Namespace.
+func CreateKnativeServing(clients *test.KnativeServingAlphaClients, names test.ResourceNames) (*v1alpha1.KnativeServing, error) {
+	ks := &v1alpha1.KnativeServing{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: names.KnativeServing,
+			Namespace: names.Namespace,
+		},
+	}
+	svc, err := clients.KnativeServings.Create(ks)
+	return svc, err
+}
+
 // WaitForKnativeServingState polls the status of the KnativeServing called name
 // from client every `interval` until `inState` returns `true` indicating it
 // is done, returns an error or timeout.
-func WaitForKnativeServingState(client *test.KnativeServingAlphaClients, name string, inState func(s *v1alpha1.KnativeServing) (bool, error)) (*v1alpha1.KnativeServing, error) {
+func WaitForKnativeServingState(client *test.KnativeServingAlphaClients, name string, inState func(s *v1alpha1.KnativeServing,
```

Comment:
i'm curious why you don't just pass `servingv1alpha1.KnativeServingInterface` in as the client. What's the value of wrapping it in `test.KnativeServingAlphaClients`?

Reply:
So far, it is the only resource KnativeServing that we will access. KnativeServingAlphaClients is designed to support multiple clients accessing multiple resources.
I am simply it.